scripts/reboot

```diff
@@ -172,22 +173,33 @@ function check_conflict_boot_in_fw_update()
 
 function parse_options()
 {
-    while getopts "h?vf" opt; do
+    while getopts "h?vftr:" opt; do
         case ${opt} in
             h|\? )
                 show_help_and_exit
                 ;;
+            f )
+                REBOOT_FORCE=true
+                echo "force: $REBOOT_FORCE"
+                ;;
             v )
                 VERBOSE=yes
+                echo "verbose: $VERBOSE"
                 ;;
             t )
                 TAG_LATEST=no
+                echo "tag_latest: $TAG_LATEST"
+                ;;
+            r )
+                REBOOT_CAUSE="$OPTARG"
+                # alternatives: $OPTARG or ${OPTARG} or "${OPTARG[@]}"
+                echo "reboot_cause: ${REBOOT_CAUSE}"
                 ;;
         esac
     done
 }
 
-parse_options $@
+parse_options "$@"
 
```

Comment:
Is this change necessary?

Reply:
Yes, this is definitely needed. In bash scripting, enclosing $@ in double-quotes is the mechanism to pick space separated string as one string, else just $@ minces it into pieces and just pick the first word from space separated string.